x-pack/test/apm_api_integration/tests/alerts/anomaly_alert.spec.ts

```diff
@@ -0,0 +1,143 @@
+
+            return [
+              ...range(0, count).flatMap((_) =>
+                serviceA
+                  .transaction('tx', 'request')
+                  .timestamp(timestamp)
+                  .duration(duration)
+                  .outcome(outcome)
+                  .serialize()
+              ),
+            ];
+          });
+
+        await synthtraceEsClient.index(events);
+        await createAndRunApmMlJob({ environment: 'production', ml });
+        const { body: createdRule } = await supertest
+          .post(`/api/alerting/rule`)
+          .set('kbn-xsrf', 'foo')
+          .send({
+            params: {
+              environment: 'production',
+              serviceName: 'service-a',
+              transactionType: 'request',
+              windowSize: 30,
+              windowUnit: 'm',
+              anomalySeverityType: 'warning',
+            },
+            consumer: 'apm',
+            schedule: {
+              interval: '1m',
+            },
+            tags: ['apm', 'service.name:service-a'],
+            name: 'Latency anomaly | service-a',
+            rule_type_id: AlertType.Anomaly,
+            notify_when: 'onActiveAlert',
+            actions: [],
+          });
+        ruleId = createdRule.id;
+      });
+
+      after(async () => {
+        await synthtraceEsClient.clean();
+        await ml.cleanMlIndices();
+        await supertest.delete(`/api/alerting/rule/${ruleId}`).set('kbn-xsrf', 'foo');
+      });
+
+      const WaitForStatusIncrement = 500;
+
+      async function waitForStatus({
+        id,
+        statuses,
+        waitMillis = 10000,
+      }: {
+        statuses: Set<string>;
+        waitMillis?: number;
+        id?: string;
+      }): Promise<Record<string, any>> {
+        if (waitMillis < 0 || !id) {
+          expect().fail(`waiting for alert ${id} statuses ${Array.from(statuses)} timed out`);
+        }
+
+        const response = await supertest.get(`/api/alerting/rule/${id}`);
+        expect(response.status).to.eql(200);
+
+        const { execution_status: executionStatus } = response.body || {};
+        const { status } = executionStatus || {};
+
+        const message = `waitForStatus(${Array.from(statuses)}): got ${JSON.stringify(
+          executionStatus
+        )}`;
+
+        if (statuses.has(status)) {
+          return executionStatus;
+        }
+
+        // eslint-disable-next-line no-console
+        console.log(`${message}, retrying`);
+
+        await delay(WaitForStatusIncrement);
+        return await waitForStatus({
+          id,
+          statuses,
+          waitMillis: waitMillis - WaitForStatusIncrement,
+        });
+      }
+
+      async function delay(millis: number): Promise<void> {
+        await new Promise((resolve) => setTimeout(resolve, millis));
+      }
+
+      it('checks if alert is active', async () => {
+        const executionStatus = await waitForStatus({ id: ruleId, statuses: new Set(['active']) });
```

Comment:
why use a `Set` here?

Reply:
no special reason, I just followed the code given as an example. I'll change it.